src/foraging_gui/GenerateMetadata.py

```diff
@@ -0,0 +1,988 @@
+        if self.behavior_streams!=[]:
+            self.session_start_time = self.behavior_streams[0].stream_start_time
+            self.session_end_time = self.behavior_streams[0].stream_end_time
+        elif self.high_speed_camera_streams!=[]:
+            self.session_start_time = self.high_speed_camera_streams[0].stream_start_time
+            self.session_end_time = self.high_speed_camera_streams[0].stream_end_time
+        elif self.ephys_streams!=[]:
+            self.session_start_time = self.ephys_streams[0].stream_start_time
+            self.session_end_time = self.ephys_streams[0].stream_end_time
+        elif self.ophys_streams!=[]:
+            self.session_start_time = self.ophys_streams[0].stream_start_time
+            self.session_end_time = self.ophys_streams[0].stream_end_time
+        else:
+            self.session_start_time = ''
+            self.session_end_time = ''
+
+    def _get_modality(self):
+        '''
+        Get all the modalities used in the session
+        '''
+        self.modality = []
+        if self.behavior_streams!=[]:
+            self.modality.append(Modality.BEHAVIOR)
+        if self.ephys_streams!=[]:
+            self.modality.append(Modality.ECEPHYS)
+        if self.ophys_streams!=[]:
+            self.modality.append(Modality.FIB)
+        if self.high_speed_camera_streams!=[]:
+            self.modality.append(Modality.BEHAVIOR_VIDEOS)
+        
+    def _get_investigators(self):
+        '''
+        Get investigators
+        '''
+        self.investigators=[]
+        investigators=self.Obj['meta_data_dialog']['session_metadata']['Investigators'].split(',')
+        for investigator in investigators:
+            if investigator != '':
+                self.investigators.append(PIDName(name=investigator, registry=self.orcid))
+
+    def _save_rig_metadata(self):
+        '''
+        Save the rig metadata to the MetadataFolder
+        '''
+        if self.Obj['meta_data_dialog']['rig_metadata_file']=='' or self.Obj['MetadataFolder']=='':
+            return
+        
+        rig_metadata_full_path=os.path.join(self.Obj['MetadataFolder'],self.Obj['meta_data_dialog']['rig_metadata_file'])
+        with open(rig_metadata_full_path, 'w') as f:
+            json.dump(self.Obj['meta_data_dialog']['rig_metadata'], f, indent=4)
+
+    def _handle_edge_cases(self):
+        '''
+        handle edge cases (e.g. missing keys in the json file)
+        '''
+        # Missing fields camera_start_time and camera_stop_time in the Camera_dialog. 
+        # Possible reason: 1) the camera is not used in the session. 2 ) the camera is used but the start and end time are not recorded for old version of the software.
+        self._initialize_fields(dic=self.Obj['Camera_dialog'],keys=['camera_start_time','camera_stop_time'],default_value='')
+        
+        # Missing Behavior data streams in the json file.
+        # Possible reason: 1) the behavior data is not started in the session. 
```

Comment:
Do we need to save metadata if there is no animal behavior? Thats a conceptual question, feel free to resolve this issue

Reply:
Either way is ok with me. 
If there are no data streams, it will save the rig metadata if it's available. The current code also permits saving other data streams without providing the behavior data stream.